src/mscorlib/shared/System/Diagnostics/Debug.Unix.cs

```diff
@@ -21,7 +21,19 @@ private static void ShowDialog(string stackTrace, string message, string detailM
                 // In Core, we do not show a dialog.
                 // Fail in order to avoid anyone catching an exception and masking
                 // an assert failure.
-                var ex = new DebugAssertException(message, detailMessage, stackTrace);
+                DebugAssertException ex;
+                if (message == String.Empty) 
+                {
+                    ex = new DebugAssertException(stackTrace);
```

Comment:
What if detailMessage contained meaningful information? Doesn't this lose that?

Reply:
This gets called from the 3 method overrides of Debug.Assert():
public static void Assert(bool condition)
public static void Assert(bool condition, string message)  
public static void Assert(bool condition, string message, string detailMessage)
and we just pass string.Empty for missing parameters... 
I suppose we would lose detailMessage if a user does something like:
```
Debug.Assert(false, String.Empty, "some detailed message");
```
While I see such case as somewhat unlikely, I do see the need to fix that for completeness so thanks for pointing that out :)